propolis/src/vmm/hdl.rs

```diff
@@ -129,12 +167,24 @@ impl VmmHdl {
         if let Some(name) = segname {
             let name_raw = name.as_bytes();
 
+            // XXX: Does this name need to be null-terminated?
+            // It's crossing an FFI boundary, and C won't have any
+            // way to distinguish the length.
```

Comment:
We start with the name array filled with zeroes, and the `< SEG_NAME_LEN` ensures that at least one NUL will remain at the end of the copied bytes.